localstripe/resources.py

```diff
@@ -939,6 +939,29 @@ def _api_list_tax_ids(cls, id, **kwargs):
         obj = cls._api_retrieve(id)
         return obj.tax_ids
 
+    @classmethod
+    def _api_retrieve_tax_id(cls, id, tax_id, **kwargs):
+        if kwargs:
+            raise UserError(400, 'Unexpected ' + ', '.join(kwargs.keys()))
+
+        obj = cls._api_retrieve(id)
+        for txi in obj.tax_ids._list:
+            if txi.id == tax_id:
+                return txi
+        raise UserError(404, 'Customer ' + id + ' does not have a TaxId with '
+                             'ID ' + tax_id)
+
+    @classmethod
+    def _api_delete_tax_id(cls, id, tax_id, **kwargs):
+        if kwargs:
+            raise UserError(400, 'Unexpected ' + ', '.join(kwargs.keys()))
+
+        obj = cls._api_retrieve(id)
+        tax_id_obj = cls._api_retrieve_tax_id(id, tax_id)
+        obj.tax_ids._list.remove(tax_id_obj)
```

Comment:
Could it be simpler, and safer, to delete the item from `obj.tax_ids._list` that matches the ID? Instead of retrieving the whole object?

Reply:
I'm not sure that it would be simpler, as we would need to:
- iterate the list 
- throw a 404 if the item is not in the list
- remove the item if `tax_id` match the id of the item
Here, the first two steps are properly handled by `_api_retrieve_tax_id`, allowing us to avoid duplicating the logic. 

I don't think that it would be safer either, as tax ID can't be modified (they can only be created or deleted), so it's not possible for a tax ID to change between `_api_retrieve_tax_id` and `obj.tax_ids._list.remove` (eg if there was a concurrent request). But maybe you had something else in mind?

Last argument: we're already doing something similar in the code: https://github.com/adrienverge/localstripe/blob/cb071d2f18fa74e1e536e9e27c0733b6c7c69b8e/localstripe/resources.py#L903-L906

Reply:
OK, I must admit all these are good arguments :+1: 

Let's just keep in mind that `.remove()` can be tricky in some cases:
```python
class O:
    def __init__(self, a):
        self.a = a

    def __repr__(self):
        return f' {self.a} '

>>> l = [O('a'), O('b'), O('c')]
>>> l.remove(l[1])
>>> print(l)
[ a ,  c ]

>>> l = [O('a'), O('b'), O('c')]
>>> l.remove(O('b'))
>>> print(l)
ValueError: list.remove(x): x not in list
```